components/authz-service/server/v2/authz.go

```diff
@@ -60,9 +66,18 @@ func (s *authzServer) ProjectsAuthorized(
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
-	// Generally we return the engine's response verbatim
-	// but there are two cases that need to be intercepted and adjusted.
-	if stringutils.SliceContains(projectsAuthorized, constants.AllProjectsID) {
+
+	// we check the version set in the channel on policy server
+	// in order to determine whether or not to filter projects for the request
+	version := s.vSwitch.version
+	if version.Minor == api.Version_V0 && len(projectsAuthorized) > 0 {
```

Comment:
i'm still mulling over the behavior here. i ran into this case while testing:

say i upgrade to v2.1 and create this policy:
```
{
  "name": "beta flag test",
  "id": "beta-test",
  "type": "CUSTOM",
  "members": [
    "token:test"
  ],
  "statements": [
    {
      "resources": [
        "*"
      ],
      "actions": [
        "iam:teams:list"
      ],
      "projects": [
        "chef",
        "automate",
        "auth"
      ]
    }
  ]
}
```

then this request always returns only `auth`, `automate`, `chef` teams
```
curl -H "api-token: $TEST" https://localhost/apis/iam/v2beta/teams --insecure | jq

{
  "teams": [
    {
      "name": "Chef",
      "id": "chef",
      "projects": [
        "chef"
      ]
    },
    {
      "name": "Automate",
      "id": "automate",
      "projects": [
        "chef",
        "automate"
      ]
    },
    {
      "name": "Auth Team",
      "id": "auth",
      "projects": [
        "chef",
        "automate",
        "auth"
      ]
    }
  ]
}
```

if i then downgrade back to v2, project filtering should no longer work. 

so my question: should this v2.1 policy apply at all? should we expand it to allow all teams? or should return no teams? 

here i've got it allowing access to ALL teams if any allowed project is returned, but now i'm thinking it might be safer to assume less permission.

Reply:
wait, why does ur example 2.1 policy have `"resources": ["*"]`?

i dont really know the best approach here, maybe a good post standup convo for tomorrow i think there are pros/cons of both approaches.

Reply:
Going from v2.1 to v2.0 is effectively saying "ignore projects" so IMO allowing access to all teams is the correct choice.